### ISSUE TITLE
CMIites - Drivers of Sales and Sales prediction - DeepTech/ML Problem Statement-3

### ISSUE THREAD

**Issue by mdave96** [skillenza-com/MishMash-India-2020#93]
**Before you start, please follow this format for your issue title**:
TEAM NAME - PROJECT NAME - THEME NAME
CMIites - Drivers of Sales and Sales prediction - DeepTech/ML Problem Statement-3

## ℹ️ Project information

 1. You can select any one theme from - XR / Mobility / FinTech/ Deep Tech or Machine Learning / Ed-Tech / Social Impact
 
- _Mobility, FinTech and Ed-Tech are Open Innovation Themes_
- _XR has 3 sub-categories that you can choose from:- Epidemic, Urban, Remake_
- _Deep Tech or Machine Learning is sponsored by UNILEVER and has 3 Problem Statements. If you pick this theme, you need to declare which problem statement you are going to work on_
- _Social Impact has only 1 Problem Statement - So, if you pick this theme you just need to select this Theme and say you will work on the Problem Statement in the Idea Brief Field_

2. **Project Name:** Give a suitable title to your project
Drivers of Sales and Sales prediction

3. **Short Project Description:** _One line crisp description of your project_
One of Unilever's brands is going through some major changes in Business Execution plans and will like to know what drives sales and how can it forecast sales for next 6 periods. 

4. **Team Name:** Please mention the same team name as mentioned over Skillenza
CMIites

5. **Team Members:** Mention their Names & tag their GitHub handles
Krishna V - krishna-venkateswaran
Pushkar Sathe - lotus745
Malhar Dave - mdave96

6. **Demo Link:** _(if any, this might contain a website/ mobile application link/ short video, etc.)_
NA

7. **Repository Link(s):** _Provide us the link to your code. All judges must be able to access it._
https://github.com/mdave96/Skillenza-Hackathon-DeepTech-ML

8. **Presentation Link:** Provide us the link to for your power point presentation. 
https://drive.google.com/open?id=16vPFj2wO29UZpnSPF5fE-axlMiA_QoxC

9. **Deep Tech - Problem Statement - 3:** If you have chosen to work on the problem statement - 3 then please submit both models based on the two datasets provided to you. 
Included in the GitHub  repo shared above

10. **Deep Tech - Problem Statement - 2:** If you have chosen to work on the problem statement - 2 then please provide the reference  for your dataset. 

11. **Azure Services Used-** Kindly mention the Azure Services used in your project. 
NA

## 🔥 Your Pitch
_Kindly write a pitch for your project. Please do not use more than 500 words_
The goal of this project was to identify what factors that drive sales and effectively use them to forecast sales 6 steps ahead.
We use different methods to arrive at the drivers of sales. This includes Gradient Boosting Regression, Bayesian Structural Time Series and Bayesian Regression.
We have used Bayesian as well as regular Machine Learning approaches and under each one, a Time Series based and a Non-Time Series based approach.
The results obtained from these approaches boast of multiple advantages such as better forecasting, better consistency etc.
For the Hurdle 1 dataset, the Vector AutoRegression (VAR) model gives the best result of 9% MAPE under hold out set.
For the hurdle 2 where data had missing values, kNN is used for imputation and Bayesian Regression gives a hold out MAPE of 10% which is quite impressive given that the data had significant no. of values missing and a smaller dataset.
In conclusion, we tried a variety of different models and achieved the set goals with great accuracy.

## 🔦 Any other specific thing you want to highlight?
We did not use any Azure services but built everything from ground up in R and Python.

## ✅ Checklist

**Before you post the issue**:
- [x] You have followed the issue title format.
- [x] You have mentioned the correct labels.
- [x] You have provided all the information correctly.
